src/platform/qt/scripting/ScriptingTextBuffer.cpp

```diff
@@ -51,26 +51,27 @@ void ScriptingTextBuffer::setBufferName(const QString& name) {
 
 void ScriptingTextBuffer::print(const QString& text) {
 	QMutexLocker locker(&m_mutex);
-	QString split(text);
-	m_shim.cursor.beginEditBlock();
-	while (m_shim.cursor.positionInBlock() + split.length() > m_dims.width()) {
-		int cut = m_dims.width() - m_shim.cursor.positionInBlock();
-		if (!m_shim.cursor.atBlockEnd()) {
-			m_shim.cursor.movePosition(QTextCursor::EndOfBlock, QTextCursor::KeepAnchor);
+	for (QString split : text.split('\n')) {
+		m_shim.cursor.beginEditBlock();
```

Comment:
Shouldn't this be outside the loop?

Reply:
I don't _think_ so? We want to create a new block for each line being inserted, don't we?

Reply:
Hm. Actually, reading the docs, why do you even need this? You don't support undo/redo in text buffers and that's all this does.

Reply:
I was using this to try to make the whole edit block "atomic" to prevent it from doing any work in the middle until it's committed. I have no idea if it works that way. Feel free to remove it if it doesn't.

Reply:
It doesn't work that way. It makes the edit block act as a single unit from an undo/redo perspective, but the thing that makes a series of edits atomic in terms of external updates is the event loop -- the various QTextDocument functions set a dirty flag on the associated QTextEdit to prompt it to redo the layout next time control returns to the event loop, much like how multiple calls to `QWidget::update()` only trigger a single `paintEvent`.

Reply:
Rebased and pushed.